### PR TITLE
Assorted Intl updates

### PR DESCRIPTION
**Pull request by jackhorton**
Going to run test262 on this locally to fix any low-hanging fruit there before merging.

### CODE REVIEW

File: lib/Runtime/Library/IntlEngineInterfaceExtensionObject.cpp [hunk 1]
```diff
@@ -3039,11 +3046,33 @@ DEFINE_ISXLOCALEAVAILABLE(PR, uloc)
...
+        SetUNumberFormatDigitOptions(*fmt, state);
+
+        char16 *formattedN = nullptr;
+        int formattedNLength = 0;
+        EnsureBuffer([&](UChar *buf, int bufLen, UErrorCode *status)
+        {
+            return unum_formatDouble(*fmt, n, buf, bufLen, nullptr, status);
+        }, scriptContext->GetRecycler(), &formattedN, &formattedNLength);
+
+        double nWithOptions = unum_parseDouble(*fmt, reinterpret_cast<UChar *>(formattedN), formattedNLength, nullptr, &status);
```
Comment (jefgen) [inline lib/Runtime/Library/IntlEngineInterfaceExtensionObject.cpp]: Perhaps check the status?

File: lib/Runtime/Library/IntlEngineInterfaceExtensionObject.cpp [hunk 2]
```diff
@@ -2747,7 +2757,6 @@ DEFINE_ISXLOCALEAVAILABLE(PR, uloc)
                 typeString = library->GetIntlLiteralPartString(); break;
 #endif
             default:
-                AssertMsg(false, "Unmapped UDateFormatField");
```
Comment (dilijev) [inline lib/Runtime/Library/IntlEngineInterfaceExtensionObject.cpp]: >AssertMsg(false, "Unmapped UDateFormatField"); [](start = 16, length = 46)

No longer needed?

File: lib/Runtime/Library/IntlEngineInterfaceExtensionObject.cpp [hunk 3]
```diff
@@ -2411,7 +2421,7 @@ DEFINE_ISXLOCALEAVAILABLE(PR, uloc)
         {
             return unum_formatDouble(*fmt, num, buf, bufLen, nullptr, status);
         }, scriptContext->GetRecycler(), &formatted, &formattedLen);
-        JavascriptOperators::InitProperty(part, PropertyIds::type, library->GetIntlLiteralPartString());
+        JavascriptOperators::InitProperty(part, PropertyIds::type, library->GetIntlUnknownPartString());
```
Comment (dilijev) [inline lib/Runtime/Library/IntlEngineInterfaceExtensionObject.cpp]: >GetIntlUnknownPartString [](start = 76, length = 24)

previously a typo or?

File: lib/Runtime/Library/IntlEngineInterfaceExtensionObject.cpp [hunk 4]
```diff
@@ -2249,8 +2257,10 @@ DEFINE_ISXLOCALEAVAILABLE(PR, uloc)
             // TODO(jahorto): Determine if this would ever be returned and what it would map to
             case UNUM_PERMILL_FIELD: AssertOrFailFastMsg(false, "Unexpected permill field");
 
-            case UNUM_SIGN_FIELD: num < 0 ? library->GetIntlMinusSignPartString() : library->GetIntlPlusSignPartString();
-            default: AssertOrFailFastMsg(false, "Unexpected unknown part"); return nullptr;
+            case UNUM_SIGN_FIELD: return num < 0 ? library->GetIntlMinusSignPartString() : library->GetIntlPlusSignPartString();
+
+            // At the ECMA-402 TC39 call for May 2017, it was decided that we should treat unmapped parts as type: "unknown"
```
Comment (dilijev) [inline lib/Runtime/Library/IntlEngineInterfaceExtensionObject.cpp]: >// At the ECMA-402 TC39 call for May 2017, it was decided that we should treat unmapped parts as type: "unknown" [](start = 12, length = 112)

oh, does this comment also apply to the other place I asked about UnkonwnPartString?

File: lib/Runtime/Library/IntlEngineInterfaceExtensionObject.cpp (hunk 3 above)
Comment (jackhorton) [reply lib/Runtime/Library/IntlEngineInterfaceExtensionObject.cpp]: Nope, this was part of the Intl call for May, where we agreed that ICU part types that we don't know how to map should be treated as `{ type: "unknown" }`. Same witht he assert removal above.

File: lib/Runtime/Library/IntlEngineInterfaceExtensionObject.cpp (hunk 4 above)
Comment (jackhorton) [reply lib/Runtime/Library/IntlEngineInterfaceExtensionObject.cpp]: Correct

**Pull request merged**